dev/io.openliberty.mcp.internal/src/io/openliberty/mcp/internal/McpSessionStore.java

```diff
@@ -0,0 +1,62 @@
+/*******************************************************************************
+ * Copyright (c) 2025 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package io.openliberty.mcp.internal;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+import jakarta.enterprise.context.ApplicationScoped;
+
+/**
+ *
+ */
+
+@ApplicationScoped
+public class McpSessionStore {
+
+    private final ConcurrentMap<String, SessionInfo> sessions = new ConcurrentHashMap<>();
+    private static final Duration SESSION_TIMEOUT = Duration.ofMinutes(10);
+
+    public String createSession() {
+        String sessionId = UUID.randomUUID().toString();
+        sessions.put(sessionId, new SessionInfo());
+        return sessionId;
+    }
+
+    public SessionInfo getSession(String sessionId) {
+        return sessions.get(sessionId);
+    }
```

Comment:
It's a bit odd that at the moment, anyone wanting the session info has to go to the `McpTransport` to get the session id, then ask the `McpSessionStore` for the actual session.

Given that we look up the sessionId on every request to validate it, it seems odd that `McpTransport` doesn't just hold the `SessionInfo` itself.